odoo/addons/base/tests/test_ir_cron.py

```diff
@@ -76,11 +76,15 @@ def setUpClass(cls):
         cls.partner = cls.env['res.partner'].create(cls._get_partner_data(cls.env))
 
     def setUp(self):
+        super().setUp()
         self.partner.write(self._get_partner_data(self.env))
         self.cron.write(self._get_cron_data(self.env))
-        self.env['ir.cron.trigger'].search(
-            [('cron_id', '=', self.cron.id)]
-        ).unlink()
+
+    def tearDown(self):
+        domain = [('cron_id', '=', self.cron.id)]
+        self.env['ir.cron.trigger'].search(domain).unlink()
+        self.env['ir.cron.progress'].search(domain).unlink()
+        super().tearDown()
```

Comment:
addCleanup inside setUp, tearDown isn't run when the test fails, addCleanup does

Reply:
I second Julien's remark.  IMO you should delete all triggers and progress records in `setUp`, so that you can analyze what's present in those tables during the test.

Reply:
Please move those two lines into `setUp()`.